tests/integration/backend_test.py

```diff
@@ -751,6 +751,42 @@ def test_device_state(
     )
 
 
+@pytest.mark.skipif(skip_remote_tests, reason=REASON)
+@pytest.mark.parametrize(
+    "authenticated_quum_backend_qa", [{"device_name": "H1-1SC"}], indirect=True
+)
+@pytest.mark.parametrize(
+    "language",
+    [
+        Language.QASM,
+        Language.QIR,
+    ],
+)
+@pytest.mark.timeout(120)
+def test_wasm_qa(
+    authenticated_quum_backend_qa: QuantinuumBackend, language: Language
+) -> None:
+    wasfile = WasmFileHandler(str(Path(__file__).parent.parent / "wasm" / "add1.wasm"))
+    c = Circuit(1)
+    c.name = "test_wasm"
+    a = c.add_c_register("a", 8)
+    c.add_wasm_to_reg("add_one", wasfile, [a], [a])
+
+    b = authenticated_quum_backend_qa
+
+    c = b.get_compiled_circuit(c)
+    h = b.process_circuits(
+        [c], n_shots=10, wasm_file_handler=wasfile, language=language  # type: ignore
+    )[0]
+
+    try:
+        assert b.get_result(h)
+    except GetResultFailed:
```

Comment:
I don't like that this test passes when the thing it's testing is not working. We should use the `xfail` annotation instead.

Reply:
I have switched this to using `xfail`

Reply:
and removed it, it is working now